addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/UsernameEnumerationScanRule.java

```diff
@@ -414,7 +408,7 @@ public void scan() {
                             new HtmlParameter(
                                     currentHtmlParameter.getType(),
                                     currentHtmlParameter.getName(),
-                                    invalidUsername.toString())); // add in the invalid username
+                                    invalidUsername)); // add in the invalid username
```

Comment:
Ditto.

Reply:
I removed a lot of comments that seemed redundant or unnecessary. There are still a lot of comments in the code but it is quite monolithic so I felt it was worth leaving them as it explains the flow and various choices.